glean-core/ios/GleanTests/DispatchersTest.swift

```diff
@@ -30,8 +30,9 @@ class DispatchersTest: XCTestCase {
             "Tasks have not executed while in queue"
         )
 
-        // Now trigger the queue to fire the tasks
-        Dispatchers.shared.flushQueuedInitialTasks()
+        // Now trigger the queue to fire the tasks, we must use the `waitUntilFinished`
+        // parameter to prevent async issues.
+        Dispatchers.shared.flushQueuedInitialTasks(waitUntilFinished: true)
```

Comment:
Does `waitUntilFinished` basically [block the caller thread](https://developer.apple.com/documentation/foundation/operationqueue/1408358-addoperations)?

If so, that's something we **do not** want to do. Please see the final version of #654

Reply:
This is JUST for the test, otherwise the foreground task executes the test assertion before the tasks have executed.

Reply:
Basically, I'm using an API feature to await the task rather than introducing some more async'ery to await the task on my own.  The actual application does not await the tasks to complete, it queues them on the background thread and moves on (see the default value for the parameter)

Reply:
Could we assert we are in testing mode if `waitUntilFinished` is `true`?  Barring that, maybe add a docstring comment that it should only be set to `true` in tests...?

Reply:
That's a GREAT idea!  Let me add that to the PR.

Reply:
I was a little less worried since this is an internal API, but it's good to document it and put up a safeguard.